root/static/scripts/edit/components/edit/RelationshipDiff.js

```diff
@@ -50,14 +51,31 @@ const makeDescriptiveLink = (
   entity: RelatableEntityT,
   content: string,
   relationship: RelationshipT,
+  showCreditedAs: boolean,
 ) => (
   <DescriptiveLink
     content={content}
     disableLink={isDisabledLink(relationship, entity)}
     entity={entity}
+    showCreditedAs={showCreditedAs}
   />
 );
 
+/*
+ * Returns true if credited-as text should be explicitly displayed
+ * alongside the entity name. This is needed when a credit that matches
+ * the entity name is being added or removed (MBS-13751).
+ */
+const shouldShowCreditedAs = (
+  entity: RelatableEntityT,
+  credit: string,
+  otherEntity: RelatableEntityT,
+  otherCredit: string,
+) => entity !== null &&
+  otherEntity !== null &&
+  entity.id === otherEntity.id &&
+  (credit === entity.name && otherCredit === '');
```

Comment:
Maybe it would be better to change this condition to `credit !== otherCredit && credit !== ''`. That way, I think we'd also call out the non-standard credit in the third edit in my example, while not cluttering the diff when the credit didn't change.

(Could also remove the parentheses here, of course, but these conditions seemed related to me. :-P)

Reply:
Shouldn't this be just `credit !== otherCredit` in that case? If both were the same or both were `''` it wouldn't show, but I guess we'd want it to show in any other case?

I expect if we change the artist but no credits are explicitly given on either side, both credits would still be `''` and nothing would be shown.

Reply:
Would `entity.id !== otherEntity.id || credit !== otherCredit` make sense here? That way, I think we'd show any credits that are present in all cases except when the entity and credit both stay the same.

Maybe it'll still be confusing if the linked text sometimes contains the credit and other times contains the entity name, though... :-/

(I think that the null checks here are also unnecessary due to the `invariant` calls in the component.)